### PR TITLE
Add `yarn generate:template` script

### PR DESCRIPTION
**Pull request by jacob-alford**
Adds `scripts/templateCombinator` and `yarn generate:template` command for generating template files for new modules.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -60,13 +61,13 @@ Schemable combinator modules must have a unique name and can be found in the fol
...
+Once the new combinator modules are in place, run `yarn generate:schemables` and the ts script will update the typeclass instances and SchemableExt module with the newly created combinators.
 
-_Coming soon: convenience function for generating new Schemable combinators._
+Create a new combinator file using `yarn generate:template [string | number] [NameOfCombinator]`. Note: the name of the new combinator must be unique, and begin with a capital letter.
 
 ### Generating Documentation
 
-Our docs pages are automatically generated whenever a PR is merged into the `main` branch, so the following step is optional. If you want to manually re-generate the documentation pages, use `yarn docs`. This command uses [docs-ts](https://github.com/gcanti/docs-ts) and JSDoc-style annotations to parse code comments for metadata about each module.
+Our docs pages are automatically generated whenever a PR is merged into the `master` branch, so the following step is optional. If you want to manually re-generate the documentation pages, use `yarn docs`. This command uses [docs-ts](https://github.com/gcanti/docs-ts) and JSDoc-style annotations to parse code comments for metadata about each module.
```
Comment (jacob-alford) [inline docs/index.md:70]: Whoops, forgot to pull dev

File: README.md [hunk 2]
```diff
@@ -54,9 +54,9 @@ Schemable combinator modules must have a unique name and can be found in the fol
 
 ### Generating SchemableExt/Instances
 
-Once the new combinator modules are in place, run `yarn generate` and the ts script will update the typeclass instances and SchemableExt module with the newly created combinators.
+Once the new combinator modules are in place, run `yarn generate:schemables` and the ts script will update the typeclass instances and SchemableExt module with the newly created combinators.
 
-_Coming soon: convenience function for generating new Schemable combinators._
+Create a new combinator file using `yarn generate:template [string | number] [NameOfCombinator]`. Note: the name of the new combinator must be unique, and begin with a capital letter.
```
Comment (newswim) [inline README.md:59]: Silly nitpick, but it might be a little more logical to switch these two paragraphs.

File: .cspell.json [hunk 3]
```diff
@@ -24,15 +24,17 @@
...
   ],
   "words": [
-    "Schemable",
+    "bigints",
     "combinators",
+    "nats",
+    "newtype",
+    "Purescript",
+    "Schemable",
+    "schemables",
```
Comment (newswim) [inline .cspell.json:33]: Neat! You'll have to tell me more about `cspell`, I've never used it but seems like a nice way of preventing typos.

File: scripts/templateCombinator.ts [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+                        _.createStringLiteral('not implemented'),
+                      ])
+                    ),
+                  ],
+                  true
+                )
+              )
+            ),
+          ],
+          ts.NodeFlags.Const
```
Comment (newswim) [inline scripts/templateCombinator.ts:180]: 🙃 why don't they make the variable signifier the first argument? 🤷

File: scripts/templateCombinator.ts [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+  path => contents => C =>
+    C.writeFile(path, contents)
+
+const format: Build<void> = C => C.exec('yarn format')
+
+const main: Build<void> = pipe(
+  process.argv,
+  RA.dropLeft(2),
+  RTE.fromPredicate(
+    (args): args is [string, string] => args.length === 2,
```
Comment (newswim) [inline scripts/templateCombinator.ts:423]: ❤️ lovely little inline refinement

File: scripts/templateCombinator.ts [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+    () =>
+      new Error(
+        'Must provide primitive type (string | number), and capitalized module name (SafeDate)'
```
Comment (newswim) [inline scripts/templateCombinator.ts:426]: 🔧 `"... module name (e.g. SafeDate)"`

File: scripts/templateCombinator.ts [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+  ),
+  RTE.filterOrElse(
+    ([, module]) => /^[A-Z]/g.test(module),
+    () => new Error('Module name must be capitalized')
+  ),
+  RTE.bindTo('args'),
+  RTE.chainFirstIOK(() => Cons.log("Checking that module doesn't already exist...")),
+  RTE.chainFirst(({ args: [, module] }) => checkModuleUniqueness(module)),
+  RTE.chainFirstIOK(() => Cons.log('Getting package version...')),
+  RTE.bind('packageVersion', () => getPackageJson),
```
Comment (newswim) [inline scripts/templateCombinator.ts:442]: Most nitpick ever.. ⛳ 

```suggestion
  RTE.apSW('packageVersion', getPackageJson)
```

**Review by newswim — APPROVED** [review]
👏 Beautiful! I see no fault, just left a couple nitpicks that you could address if you think they're worthwhile.

File: scripts/templateCombinator.ts (hunk 7 above)
Comment (jacob-alford) [reply scripts/templateCombinator.ts:442]: No, you're totally right, this is an apS moment

File: .cspell.json (hunk 3 above)
Comment (jacob-alford) [reply .cspell.json:33]: Yeah!  It works with the [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) vscode extension.  And we have a yarn command for it: `yarn spelling`

File: scripts/ts-helpers.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      undefined,
+      _.createNamespaceImport(_.createIdentifier(name))
+    ),
+    _.createStringLiteral(path),
+    undefined
+  )
+
+export const makeDestructureImport: (
+  destructuredElements: ReadonlyArray<string>,
+  path: string
```
[the remark this answers is not in the document]
Comment (jacob-alford) [reply scripts/ts-helpers.ts:24]: Not sure if these need tests tbh, don't think it's all that worth it to add tests for scripts

File: scripts/templateCombinator.ts (hunk 4 above)
Comment (jacob-alford) [reply scripts/templateCombinator.ts:180]: That's a great question 🤔

**Review by newswim — APPROVED** [review]
👏 Lovely!

**Pull request merged**